ceno_zkvm/src/expression.rs

```diff
@@ -767,48 +776,55 @@ impl WitIn {
 
 pub trait ToExpr<E: ExtensionField> {
     type Output;
-    fn expr(&self) -> Self::Output;
+    fn expr(self) -> Self::Output;
+}
+
+impl<E: ExtensionField> ToExpr<E> for Expression<E> {
+    type Output = Expression<E>;
+    fn expr(self) -> Self::Output {
+        self
+    }
+}
+
+impl<E: ExtensionField> ToExpr<E> for &Expression<E> {
+    type Output = Expression<E>;
+    fn expr(self) -> Self::Output {
+        self.clone()
+    }
 }
 
 impl<E: ExtensionField> ToExpr<E> for WitIn {
     type Output = Expression<E>;
-    fn expr(&self) -> Expression<E> {
+    fn expr(self) -> Self::Output {
         Expression::WitIn(self.id)
     }
 }
 
 impl<E: ExtensionField> ToExpr<E> for &WitIn {
     type Output = Expression<E>;
-    fn expr(&self) -> Expression<E> {
+    fn expr(self) -> Self::Output {
         Expression::WitIn(self.id)
     }
 }
 
 impl<E: ExtensionField> ToExpr<E> for Fixed {
     type Output = Expression<E>;
-    fn expr(&self) -> Expression<E> {
-        Expression::Fixed(*self)
+    fn expr(self) -> Self::Output {
+        Expression::Fixed(self)
     }
 }
 
 impl<E: ExtensionField> ToExpr<E> for &Fixed {
     type Output = Expression<E>;
-    fn expr(&self) -> Expression<E> {
-        Expression::Fixed(**self)
+    fn expr(self) -> Self::Output {
+        Expression::Fixed(*self)
     }
 }
 
 impl<E: ExtensionField> ToExpr<E> for Instance {
     type Output = Expression<E>;
-    fn expr(&self) -> Expression<E> {
-        Expression::Instance(*self)
-    }
-}
-
-impl<F: SmallField, E: ExtensionField<BaseField = F>> ToExpr<E> for F {
```

Comment:
This implementation of the trait is a bit problematic, because it clashes with making implementations for eg `u64`.